src/cpp/log/FileConsumer.cpp

```diff
@@ -0,0 +1,77 @@
+#include <fastrtps/log/FileConsumer.h>
+#include <iomanip>
+
+namespace eprosima {
+namespace fastrtps {
+
+FileConsumer::FileConsumer()
+    : FileConsumer("output.log")
+{
+}
+
+FileConsumer::FileConsumer(const std::string &filename, bool append)
+    : mOutputFile(filename)
+    , mAppend(append)
+{
+    if (append)
+    {
+        mFile.open(mOutputFile, std::ios::out | std::ios::app);
+    }
+    else
+    {
+        mFile.open(mOutputFile, std::ios::out);
+    }
+}
+
+FileConsumer::~FileConsumer()
+{
+    mFile.close();
+}
+
+void FileConsumer::Consume(const Log::Entry& entry)
+{
+   PrintHeader(entry);
+   mFile << entry.message;
+   PrintContext(entry);
+   mFile << std::endl;
+   mFile.flush();
+}
+
+void FileConsumer::PrintHeader(const Log::Entry& entry)
+{
+    auto now = std::chrono::system_clock::now();
```

Comment:
This code is common with Stdconsumer. Cannot be in the parent (LogConsumer)?

Reply:
I have merged the timestamp block into a common method in the parent class